server/src/test/java/org/candlepin/dto/rules/v1/PoolTranslatorTest.java

```diff
@@ -75,19 +79,15 @@ protected Pool initSourceObject() {
         derivedProduct.setAttribute(Product.Attributes.STACKING_ID, "2221");
         source.setDerivedProduct(derivedProduct);
 
-        ProvidedProduct providedProd = new ProvidedProduct();
-        providedProd.setProductId("provided-product-id-1");
-        providedProd.setProductName("provided-product-name-1");
-        Set<ProvidedProduct> providedProducts = new HashSet<>();
-        providedProducts.add(providedProd);
-        source.setProvidedProductDtos(providedProducts);
+        Product providedProd = new Product();
+        providedProd.setId("provided-product-id-1");
+        providedProd.setName("provided-product-name-1");
+        source.getProduct().addProvidedProduct(providedProd);
 
-        ProvidedProduct derivedProvidedProd = new ProvidedProduct();
-        derivedProvidedProd.setProductId("derived-provided-product-id-1");
-        derivedProvidedProd.setProductName("derived-provided-product-name-1");
-        Set<ProvidedProduct> derivedProvidedProducts = new HashSet<>();
-        derivedProvidedProducts.add(derivedProvidedProd);
-        source.setDerivedProvidedProductDtos(derivedProvidedProducts);
+        Product derivedProvidedProd = new Product();
+        derivedProvidedProd.setId("derived-provided-product-id-1");
+        derivedProvidedProd.setName("derived-provided-product-name-1");
+        source.getDerivedProduct().addProvidedProduct(derivedProvidedProd);
```

Comment:
Add this directly to the derived product instead.

Reply:
Corrected the adding provided product to product.